field_utils/FieldBundleDelta.F90

```diff
@@ -0,0 +1,306 @@
+         call MAPL_InfoGetInternal(bundle_b, key=KEY_INTERPOLATION_WEIGHTS, values=weights_b, _RC)
+
+         if (any(weights_a /= weights_b)) then
+            interpolation_weights = weights_b
+         end if
+
+         _RETURN(_SUCCESS)
+
+      end subroutine compute_interpolation_weights_delta
+
+      subroutine compute_field_delta(field_delta, bundle_a, bundle_b, rc)
+         type(FieldDelta), intent(out) :: field_delta
+         type(ESMF_FieldBundle), intent(in) :: bundle_a
+         type(ESMF_FieldBundle), intent(in) :: bundle_b
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+         integer :: fieldCount_a, fieldCount_b
+         type(ESMF_Field), allocatable :: fieldList_a(:), fieldList_b(:)
+
+         call ESMF_FieldBundleGet(bundle_a, fieldCount=fieldCount_a, _RC)
+         call ESMF_FieldBundleGet(bundle_b, fieldCount=fieldCount_b, _RC)
+         allocate(fieldList_a(fieldCount_a), fieldList_b(fieldCount_b))
+         
+         if ((fieldCount_a > 0) .and. (fieldCount_b > 0)) then
+            call ESMF_FieldBundleGet(bundle_a, fieldList=fieldList_a, _RC)
+            call ESMF_FieldBundleGet(bundle_b, fieldList=fieldList_b, _RC)
+            call field_delta%initialize(fieldList_a(1), fieldList_b(1), _RC)
+            _RETURN(_SUCCESS)
+         end if
+
+         if (fieldCount_b > 0) then
+            call ESMF_FieldBundleGet(bundle_b, fieldList=fieldList_b, _RC)
+            ! full FieldDelta
+            call field_delta%initialize(fieldList_b(1), _RC)
+            _RETURN(_SUCCESS)
+         end if
+
+         ! Otherwise nothing to do. Fields are either going away
+         ! (n_fields_b = 0) or there are no fields on either side
+         ! (n_fields_a = 0 and n_fields_b = 0).
+            
+         _RETURN(_SUCCESS)
+      end subroutine compute_field_delta
+      
+
+   end subroutine initialize_bundle_delta
+
+   subroutine update_bundle(this, bundle, ignore, rc)
+      class(FieldBundleDelta), intent(in) :: this
+      type(ESMF_FieldBundle), intent(inout) :: bundle
+      character(*), intent(in), optional :: ignore
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      character(:), allocatable :: ignore_
+      type(ESMF_Field), allocatable :: fieldList(:)
+
+      ignore_ = ''
+      if (present(ignore)) ignore_ = ignore
+
+      call this%reallocate_bundle(bundle, ignore=ignore_, _RC)
+      call MAPL_FieldBundleGet(bundle, fieldList=fieldList, _RC)
+      call this%field_delta%update_fields(fieldList, ignore=ignore_, _RC)
+
+      ! unique attribute in bundle
+      call update_interpolation_weights(this%interpolation_weights, bundle, ignore=ignore_, _RC)
+
+      _RETURN(_SUCCESS)
+   contains
+
+      subroutine update_units(units, field, ignore, rc)
+         character(*), optional, intent(in) :: units
+         type(ESMF_Field), intent(inout) :: field
+         character(*), intent(in), optional :: ignore
```

Comment:
```suggestion
         character(*), intent(in) :: ignore
```

Reply:
Outer procedure constructs a local variable that is always "present", so "optional" is misleading here.